File: templates/problems/07_trees/01_invert_tree/invert_tree_test.go
package invert_tree

import (
	"reflect"
	"testing"
)

func TestInvertTree(t *testing.T) {
	tests := []struct {
		name string
		vals []int
		want []int
	}{
		{"example 1", []int{4, 2, 7, 1, 3, 6, 9}, []int{4, 7, 2, 9, 6, 3, 1}},
		{"example 2", []int{2, 1, 3}, []int{2, 3, 1}},
		{"empty tree", []int{}, []int{}},
		{"single node", []int{1}, []int{1}},
		{"left only", []int{1, 2}, []int{1, -101, 2}},
		{"right only", []int{1, -101, 2}, []int{1, 2}},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			root := buildTreeHelper(tt.vals)
			got := treeToSlice(invertTree(root))
			if !reflect.DeepEqual(got, tt.want) {
				t.Errorf("invertTree() = %v, want %v", got, tt.want)
			}
		})
	}
}

File: templates/problems/07_trees/01_invert_tree/typescript/invert_tree.test.ts
import { describe, it, expect } from "vitest";
import { buildTree, treeToSlice } from "./helpers";
import { invertTree } from "./invert_tree";

describe("invertTree", () => {
    it.each([
        { name: "example 1", vals: [4, 2, 7, 1, 3, 6, 9], want: [4, 7, 2, 9, 6, 3, 1] },
        { name: "example 2", vals: [2, 1, 3], want: [2, 3, 1] },
        { name: "empty tree", vals: [], want: [] },
        { name: "single node", vals: [1], want: [1] },
        { name: "left only", vals: [1, 2], want: [1, -101, 2] },
        { name: "right only", vals: [1, -101, 2], want: [1, 2] },
    ])("case: $name", ({ vals, want }) => {
        const root = buildTree(vals);
        expect(treeToSlice(invertTree(root))).toEqual(want);
    });
});
